pkg/kando/location.go

```diff
@@ -38,17 +53,119 @@ func newLocationCommand() *cobra.Command {
 	cmd.AddCommand(newLocationDeleteCommand())
 	cmd.PersistentFlags().StringP(pathFlagName, "s", "", "Specify a path suffix (optional)")
 	cmd.PersistentFlags().StringP(profileFlagName, "p", "", "Pass a Profile as a JSON string (required)")
-	_ = cmd.MarkFlagRequired(profileFlagName)
+	cmd.PersistentFlags().StringP(repositoryServerFlagName, "r", "", "Pass a Repository Server CR as a JSON string (required for kopia based blueprints)")
 	return cmd
 }
 
+// NewDataMover creates an instance of DataMover Interface and returns
+// the preferred DataMover as per the arguments passed in kando command
+func NewDataMover(dm *DataMoverArgs, outputName, snapJson string) datamover.DataMover {
+	switch dm.Type {
```

Comment:
As mentioned above, I am not sure we really type here. You should be able to determine the type by checking if `Profile` or `RepositoryServer` flags are set directly.

Please go through some interface examples to see if there's a simpler way to organize this.

Reply:
Refactored the code !